src/sql-parser/src/parser.rs

```diff
@@ -2314,15 +2341,21 @@ impl<'a> Parser<'a> {
                 }))
             }
             KINESIS => {
+                self.expect_keyword(CONNECTION)?;
+                let connection = self.parse_raw_name()?;
+
                 self.expect_keyword(ARN)?;
                 let arn = self.parse_literal_string()?;
-                Ok(CreateSourceConnection::Kinesis { arn })
+                Ok(CreateSourceConnection::Kinesis { connection, arn })
             }
             S3 => {
-                // FROM S3 DISCOVER OBJECTS
+                // FROM S3 CONNECTION <s3 CONNECTION> DISCOVER OBJECTS
```

Comment:
```suggestion
                // FROM S3 CONNECTION <aws CONNECTION> DISCOVER OBJECTS
```
Seems important to make clear that we have `AWS` connections, that can be used by multiple different sources